packetbeat/protos/amqp/amqp_test.go

```diff
@@ -0,0 +1,1083 @@
+		"17800ce02000100000018003c0000000000000000003a800009696d6167652f676966" +
+		"ce0300010000003a41414141414141414141414141414141414141414141414141414141" +
+		"414141414141414141414141414141414141414141414141414141414141ce")
+	assert.Nil(t, err)
+
+	tcptuple = testTcpTuple()
+
+	req = protos.Packet{Payload: data}
+	private = protos.ProtocolData(new(amqpPrivateData))
+
+	//method frame
+	private = amqp.Parse(&req, tcptuple, 0, private)
+
+	trans = expectTransaction(t, amqp)
+
+	assert.Equal(t, "basic.publish", trans["method"])
+	assert.Equal(t, "amqp", trans["type"])
+	assert.Equal(t, "65 65 65 65 65 65 65 65 65 65 [...]", trans["request"])
+	assert.Equal(t, common.OK_STATUS, trans["status"])
+	fields, ok = trans["amqp"].(common.MapStr)
+	if !ok {
+		t.Errorf("Field should be present")
+	}
+	assert.Equal(t, "image/gif", fields["content-type"])
+	assert.Equal(t, "TestMax", fields["routing-key"])
+	assert.Equal(t, false, fields["immediate"])
+	assert.Equal(t, false, fields["mandatory"])
+	assert.Nil(t, fields["exchange"])
+}
+
+func TestAmqp_HideArguments(t *testing.T) {
+
+	amqp := AmqpModForTests()
+	amqp.Send_request = true
+	amqp.Parse_headers = false
+	amqp.Parse_arguments = false
+
+	//parse args
+	data, err := hex.DecodeString("0100010000004d0032000a00000a5465737448656164" +
+		"6572180000003704626f6f6c74010362697462050568656c6c6f530000001f4869206461" +
+		"726c696e6720c3aac3aac3aac3aac3aac3aac3aae697a5e69cacce")
+	assert.Nil(t, err)
+	tcptuple := testTcpTuple()
+	req := protos.Packet{Payload: data}
+	private := protos.ProtocolData(new(amqpPrivateData))
+	private = amqp.Parse(&req, tcptuple, 0, private)
+
+	trans := expectTransaction(t, amqp)
+	assert.Equal(t, "queue.declare", trans["method"])
+	assert.Equal(t, "amqp", trans["type"])
+	assert.Equal(t, "queue.declare TestHeader", trans["request"])
+	fields, ok := trans["amqp"].(common.MapStr)
+	if !ok {
+		t.Errorf("Field should be present")
+	}
+	assert.Equal(t, false, fields["durable"])
+	assert.Equal(t, true, fields["auto-delete"])
+	_, exists := fields["arguments"].(common.MapStr)
+	if exists {
+		t.Errorf("Arguments field should not be present")
+	}
+
+	//parse headers
+	data, err = hex.DecodeString("01000100000013003c00280000000a546573744865616" +
+		"4657200ce02000100000026003c0000000000000000001a98800a746578742f706c61696" +
+		"e02060a656c206d656e73616a65ce0300010000001a54657374206865616465722066696" +
+		"56c647320666f7265766572ce")
+	assert.Nil(t, err)
+	tcptuple = testTcpTuple()
+	req = protos.Packet{Payload: data}
+	private = protos.ProtocolData(new(amqpPrivateData))
+	private = amqp.Parse(&req, tcptuple, 0, private)
+	trans = expectTransaction(t, amqp)
+	assert.Equal(t, "basic.publish", trans["method"])
+	assert.Equal(t, "amqp", trans["type"])
+	fields, ok = trans["amqp"].(common.MapStr)
+	if !ok {
+		t.Errorf("Field should be present")
+	}
+	assert.Equal(t, "TestHeader", fields["routing-key"])
+	assert.Nil(t, fields["exchange"])
```

Comment:
Here the assert.Nil(t, fields["exchange"]) is there to verify that the "exchange" field is not present.